packages/core/src/factory.ts

```diff
@@ -62,18 +63,28 @@ export function factory<Resource, Input, Output, Config extends IPrismConfig>(
     config: Config,
     validations: IPrismDiagnostic[]
   ): TE.TaskEither<Error, ResourceAndValidation & { output: Output }> => {
-    const mockCall = components.mock({
-      resource,
-      input: { data, validations },
-      config: config.mock,
-    });
+    const mockCall: IOE.IOEither<Error, Output> = () =>
```

Comment:
Why it is now effectful?

Reply:
You are right, I typed this incorrectly (although the types match). Will fix.

Reply:
Removed type annotation in 0a68cd1